examples/fabric-admin/rpc/RpcServer.cpp

```diff
@@ -62,6 +63,43 @@ class FabricAdmin final : public rpc::FabricAdmin
         return pw::OkStatus();
     }
 
+    pw::Status ReverseCommissionBridge(const chip_rpc_DeviceCommissioningInfo & request, pw_protobuf_Empty & response) override
+    {
+        char saltHex[Crypto::kSpake2p_Max_PBKDF_Salt_Length * 2 + 1];
+        Encoding::BytesToHex(request.salt.bytes, request.salt.size, saltHex, sizeof(saltHex), Encoding::HexFlags::kNullTerminate);
+
+        ChipLogProgress(NotSpecified, "Received ReverseCommissionBridge request");
+
+        SetupPayload setupPayload = SetupPayload();
+
+        setupPayload.setUpPINCode = request.setup_pin;
+        setupPayload.version      = 0;
+        setupPayload.rendezvousInformation.SetValue(RendezvousInformationFlag::kOnNetwork);
+
+        SetupDiscriminator discriminator{};
+        discriminator.SetLongValue(request.discriminator);
+        setupPayload.discriminator = discriminator;
+
+        char payloadBuffer[kMaxManaulCodeLength + 1];
+        MutableCharSpan manualCode(payloadBuffer);
+
+        CHIP_ERROR error = ManualSetupPayloadGenerator(setupPayload).payloadDecimalStringRepresentation(manualCode);
+        if (error == CHIP_NO_ERROR)
+        {
+            NodeId nodeId = DeviceMgr().GetNextAvailableNodeId();
+
+            usleep(kCommissionPrepareTimeMs * 1000);
```

Comment:
Can we have a comment on why this sleep is needed? Shouldn't the node we are commissioning be ready so I don't quite understand what we are waiting for here

Reply:
Acked

Reply:
What happens if you don't wait. I know locally I have been able to start trying to commission a device before device is comissionable. A couple seconds later I made the device commissionable and everything was fine

Reply:
It will eventually go through, but based on my observations, the success rate is lower, w/o the delay.